### PR TITLE
Use astroquery to make the VP

### PR DESCRIPTION
**Pull request by DanielAndreasen**
There are still some problems here. The two extra y-axis doesn't show, we can't use LaTeX syntax, and for some reason there is a problem with one of the yticklabels (see the `TODO`).

### CODE REVIEW

File: visibilityPlot.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+from matplotlib.font_manager import FontProperties
+from matplotlib import rcParams
+rcParams['xtick.major.pad'] = 12
+font0 = FontProperties()
+font1 = font0.copy()
+font0.set_family('sans-serif')
+font0.set_weight('light')
+font1.set_family('sans-serif')
+font1.set_weight('medium')
+from mpld3 import fig_to_html, plugins
```
Comment (j-faria) [inline visibilityPlot.py:21]: `mpld3` is now unmaintained... :(

File: visibilityPlot.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    utchours = (np.arange(0,72,dtype=int)+12)%24
+  else:
+    jdhours = np.arange(0,3,1.0/12.)
+    utchours = (np.arange(0,72, 2, dtype=int)+12)%24
+  ax.set_xticks(jdhours)
+  ax.set_xlim(axrange)
+  ax.set_xticklabels(utchours, fontsize=18)
+
+  # Make ax2 responsible for "top" axis and "right" axis
+  ax2 = ax.twin()
```
Comment (j-faria) [inline visibilityPlot.py:162]: Apparently, the twin axis are actually there but on top of each other, see [this issue](https://github.com/mpld3/mpld3/issues/188)

File: visibilityPlot.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+  ax.set_ylim([0, 91])
+  ax.yaxis.set_major_locator(MultipleLocator(15))
+  ax.yaxis.set_minor_locator(MultipleLocator(5))
+  yticks = ax.get_yticks()
+  ytickformat = []
+  for t in range(yticks.size):
+      ytickformat.append(str(int(yticks[t]))+r"$^\circ$")
+
+  #TODO: This is where it breaks
```
Comment (j-faria) [inline visibilityPlot.py:224]: I would say forget about the twin axis

File: visibilityPlot.py [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+  for t in range(yticksminor.size):
+      m_ytickformat.append(str(int(yticksminor[t]))+r"$^\circ$")
+  ax.set_yticklabels(m_ytickformat, minor=True)
+  ax.set_ylim([0, 91])
+
+  ax.yaxis.grid(color='gray', linestyle='dashed')
+  ax.yaxis.grid(color='gray', which="minor", linestyle='dotted')
+  ax2.xaxis.grid(color='gray', linestyle='dotted')
+
+  plt.text(0.5,0.95,"Visibility on {0!s}".format(date.date()), \
```
Comment (j-faria) [inline visibilityPlot.py:244]: We can render this text in the html page itself and save some work on the Python/matplotlib side

File: visibilityPlot.py [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+    line3 = matplotlib.lines.Line2D((0,0),(1,1), color='k', linestyle="-", linewidth=2)
+
+    lgd2 = plt.legend((line1,line2,line3),("day","twilight","night",), \
+                        bbox_to_anchor=(0.88, 0.13), loc='best', borderaxespad=0.,prop={'size':12}, fancybox=True)
+    lgd2.get_frame().set_alpha(.5)
+
+  obsco = r"Obs coord.: {0:8.4f}$^\circ$, {1:8.4f}$^\circ$, {2:4f} m".format(obs['longitude'], obs['latitude'], obs['altitude'])
+
+  plt.text(0.01,0.97, obsco, transform=fig.transFigure, ha='left', va='center', fontsize=10)
+  plt.text(0.01,0.95, obs['name'], transform=fig.transFigure, ha='left', va='center', fontsize=10)
```
Comment (j-faria) [inline visibilityPlot.py:259]: same here

File: app.py [hunk 6]
```diff
@@ -1,27 +1,30 @@
...
-    print(text)
-    processed_text = text['target'].upper()
-    return processed_text
+    form = request.form
+    date = dt.datetime(*[int(i) for i in form['date'].split('-')])
+    target = form['target']
+    observatory = obs[form['observatory']]
+
+    plot = visibilityPlot(date=date, target=target, observatory=observatory)
+    return render_template('plot.html', plot=plot)
```
Comment (j-faria) [inline app.py:27]: It will be easy to provide `plot.html` with the date / target / observatory combo, so it can render this info on the html itself.

**Review by j-faria — COMMENTED** [review]
Hi Daniel,
Thanks a lot for working on this! I wouldn't have any idea where to start.

I'm not sure how far we'll be able to go with mpld3, so my suggestion would be to keep what works in the plot (the main axis with the visibility curve) and later start more fresh with another library (bokeh?).
Also let's render the text parts in the page itself. 

Overall, I think the hard part here is done: take input from the forms, plot the visibility curve with those inputs! Great work!

**Comment by DanielAndreasen** [comment]
Hi João,

Always fun to do this kind of stuff when I have time. Now the ground work is made, and we can continue from here. I choosed `mpld3` here because I didn't have to make any effort making it work.
We can look into using another library like bokeh, but that will obviously take some time.

Good idea about rendering text directly in the html. Not sure when I will have time to work on it again though :/

**Comment by DanielAndreasen** [comment]
Are there any plans to use this? Otherwise we can just close the PR.

**Pull request closed without merging**